### ISSUE TITLE
build error on fl2000_surface.c

### ISSUE THREAD

**Issue by C-Bouthoorn** [FrescoLogic/FL2000#6]
```
/FL2000/src/fl2000_surface.c: In function ‘fl2000_surface_pin_down’:
/FL2000/src/fl2000_surface.c:182:3: error: implicit declaration of function ‘page_cache_release’; did you mean ‘pagecache_get_page’? [-Werror=implicit-function-declaration]
   page_cache_release(pages[i]);
   ^~~~~~~~~~~~~~~~~~
   pagecache_get_page
```

I used ```KERNEL_PATH = /usr/lib/modules/`uname -r`/build``` in the Makefile.

`uname -r` gives `4.12.8-2-ARCH`

I have the `linux-headers` package installed (version 4.12.8-2)

**Comment by midaszhou** [comment]
Try to use  _**put_page(pages[i],0)**_  instead.

**Comment by fresco-fl2000** [comment]
I've fixed the compiler errors on kernel 4.11.0, and usb_control_msg() failure. Starting from some kernel version, the USB core discriminates against data buffer from stack (eg. local variables) when using usb_control_transfer.

The following line shows the core team decides that buffer  allocated from vmalloc() is banned when the client driver wishes to call usb_control_msg(). This is really rediculous.

http://elixir.free-electrons.com/linux/v3.18/source/drivers/usb/core/hcd.c#L1503

anyway, please try the latest code and see if it helps.
